libraries/apollo-ast/api/apollo-ast.api

```diff
@@ -968,17 +1010,38 @@ public final class com/apollographql/apollo3/ast/TransformResult$Replace : com/a
 	public final fun getNewNode ()Lcom/apollographql/apollo3/ast/GQLNode;
 }
 
+public final class com/apollographql/apollo3/ast/UnknownDirective : com/apollographql/apollo3/ast/GraphQLValidationIssue {
+	public fun <init> (Ljava/lang/String;Lcom/apollographql/apollo3/ast/SourceLocation;)V
+	public fun getMessage ()Ljava/lang/String;
+	public fun getSourceLocation ()Lcom/apollographql/apollo3/ast/SourceLocation;
+}
+
 public final class com/apollographql/apollo3/ast/UnrecognizedAntlrRule : com/apollographql/apollo3/ast/SourceAwareException {
 	public fun <init> (Ljava/lang/String;Lcom/apollographql/apollo3/ast/SourceLocation;)V
 }
 
-public final class com/apollographql/apollo3/ast/ValidationDetails : java/lang/Enum {
-	public static final field DuplicateTypeName Lcom/apollographql/apollo3/ast/ValidationDetails;
-	public static final field Other Lcom/apollographql/apollo3/ast/ValidationDetails;
-	public static final field UnknownDirective Lcom/apollographql/apollo3/ast/ValidationDetails;
-	public static fun getEntries ()Lkotlin/enums/EnumEntries;
-	public static fun valueOf (Ljava/lang/String;)Lcom/apollographql/apollo3/ast/ValidationDetails;
-	public static fun values ()[Lcom/apollographql/apollo3/ast/ValidationDetails;
```

Comment:
There are a bunch of breaking changes in the AST API. I expect them to be mostly harmless. Apollo AST in 3.x was mostly in preview.